### PR TITLE
[GH-283]: Fixed issue #283 'Removed the code for depreciated JWT'

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
**Summary** 
- Zoom has depreciated its JWT authentication offering so we have also formally deprecated the JWT offering and removed it from the code.

- [Issue #283](https://github.com/mattermost/mattermost-plugin-zoom/issues/283)

### CODE REVIEW

File: server/configuration.go [hunk 1]
```diff
@@ -56,16 +51,7 @@ func (c *configuration) Clone() *configuration {
...
-	case !c.EnableOAuth && !isCloud: // JWT for on-prem
-		switch {
-		case len(c.APIKey) == 0:
-			return errors.New("please configure APIKey")
-
-		case len(c.APISecret) == 0:
-			return errors.New("please configure APISecret")
-		}
-	case c.EnableOAuth || isCloud: // OAuth for either platform
+	if isCloud { // OAuth for either platform
```
Comment (mickmister) [inline server/configuration.go:54]: We want to just remove this if condition, and have the block run unconditionally

```suggestion
```

**Review by mickmister — CHANGES_REQUESTED** [review]
Just one request, otherwise LGTM :+1:

File: server/configuration.go (hunk 1 above)
Comment (mickmister) [reply server/configuration.go]: As a follow up explanation here, every other condition removed in this PR is `OAuthEnabled() &&`, whereas this one is `OAuthEnabled() ||` so we treat it slightly differently.

**Review by mickmister — APPROVED** [review]

File: docs/installation/mattermost-setup.md [hunk 2]
```diff
@@ -30,10 +30,5 @@ description: Configuration steps for the Mattermost server
     * Select **Regenerate** next to the field **At Rest Token Encryption Key**.
     * Make sure **Enable Password based authentication** is set to `false`.
     * Ignore **API Key** and **API Secret** fields.
-  * **JWT/Password?**
-    * Make sure **Enable OAuth** is set to `false`.
-    * Ignore the fields **Zoom OAuth Client ID**, **Zoom OAuth Client Secret**, and **At Rest Token Encryption Key**.
-    * Set **Enable Password based authentication** to `true`.
-    * Use the API Key and API Secret generated during [Zoom Configuration](https://github.com/mattermost/mattermost-plugin-zoom/tree/8cc104ee70b0b51ac7e4c8c73715f25c42c782e0/docs/installation/zoom-configuration/zoom-setup-jwt.md) to fill in the fields **API Key** and **API Secret**.
 * If you are using Webhooks or Deauthorization, make sure you hit the **Regenerate** button on **Webhook Secret** field.
```
Comment (hanzei) [inline docs/installation/mattermost-setup.md:33]: Given that OAuth is now the only way to connect, do we even need the `How are your users connecting to Zoom?` paragraph?

File: server/zoom/oauth.go [hunk 3]
```diff
@@ -179,7 +180,6 @@ func (c *OAuthClient) getUserViaOAuth(user *model.User, firstConnect bool) (*Use
 			c.token = updatedToken
 		}
 	}
-
```
Comment (hanzei) [inline server/zoom/oauth.go:182]: Nit: unrelated change

**Review by hanzei — CHANGES_REQUESTED** [review]
LGTM, one blocking documentation comment

**Comment by hanzei** [comment]
@mgdelacroix I'm going to remove you from the reviewers list as you seem too busy to review it currently.

**Review by hanzei — APPROVED** [review]

**Comment by hanzei** [comment]
@DHaussermann Ready for your review

**Comment by mickmister** [comment]
Note that we've merged a PR https://github.com/mattermost/mattermost-plugin-zoom/pull/310 that removes JWT from the readme, which had some merge conflicts here

**Comment by avas27JTG** [comment]
Fixes https://github.com/mattermost/mattermost-plugin-zoom/issues/283

**Comment by AayushChaudhary0001** [comment]
Tested and Approved

This PR is tested for the following scenarios:-

- Change in the plugin config settings
- Updated readme.md file

All changes seems fine and are working perfect, LGTM

**Review by AayushChaudhary0001 — APPROVED** [review]

**Pull request merged**